src/librustdoc/scrape_examples.rs

```diff
@@ -240,6 +252,13 @@ crate fn run(
         let mut finder = FindCalls { calls: &mut calls, tcx, map: tcx.hir(), cx, target_crates };
         tcx.hir().visit_all_item_likes(&mut finder.as_deep_visitor());
 
+        // Sort call locations within a given file in document order
+        for fn_calls in calls.values_mut() {
+            for file_calls in fn_calls.values_mut() {
+                file_calls.locations.sort_by_key(|loc| loc.call_expr.byte_span.0);
+            }
+        }
+
         // Save output to provided path
```

Comment:
Are you sure the order is preserved when serializing and deserializing? Maybe it makes sense to sort it during deserialization instead?

Please add a test for this either way.

Reply:
I'm assuming that the serialization library would serialize / deserialize a vector in the order as given (but not, say a hashmap).

I updated the `rustdoc-scrape-examples-ordering` test to check that the `focus` class is applied to the first example in a file.